rust/src/io/commit.rs

```diff
@@ -88,20 +96,22 @@ async fn list_manifests<'a>(
 }
 
 async fn write_latest_manifest(
-    commit_handler: &dyn CommitHandler,
     from_path: &Path,
     base_path: &Path,
     object_store: &ObjectStore,
 ) -> Result<()> {
-    let latest_path = commit_handler
-        .resolve_latest_version(base_path, object_store)
-        .await?;
+    let latest_path = latest_manifest_path(base_path);
     // Then, create a copy at the latest manifest path
```

Comment:
nit: this comment feels a little about of place in the sequence. Should it be moved down a few lines?